include/BoardData.hxx

```diff
@@ -33,11 +33,17 @@
 #include <vector>
 #include "utility.hxx"
 
+/**
+* Protects shared data from being simultaneously accessed by multiple threads
+**/
 struct BidirMutexes {
     std::mutex rx_mutex;
     std::mutex tx_mutex;
 };
 
+/**
+* Contains the fixed size of the buffer from transmitter and reciever and its used size
```

Comment:
do you mean "it's used by size"

Reply:
Well isn't it's -> it is, and its means it belongs to something. If i am not mistaken

Reply:
never mind, I get it now